### PR TITLE
test/e2e: add HTTPProxy 007,018 and Ingress 001 tests

### PR DESCRIPTION
**Pull request by skriss**
Updates #3621.

Signed-off-by: Steve Kriss <krisss@vmware.com>

### CODE REVIEW

File: test/e2e/httpproxy/007_client_cert_auth_test.go [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+			host:       authProxy.Spec.VirtualHost.Fqdn,
+			clientCert: &invalidClientCert,
+			wantErr:    "tls: certificate required",
+		},
+
+		"echo-with-auth-skip-verify without a client cert should succeed": {
+			host:       authSkipVerifyProxy.Spec.VirtualHost.Fqdn,
+			clientCert: nil,
+			wantErr:    "",
+		},
```
Comment (skriss) [inline test/e2e/httpproxy/007_client_cert_auth_test.go:385]: @sunjayBhatia this is the case we were discussing.

**Comment by skriss** [comment]
@sunjayBhatia ping for a review when you get a chance, thanks!

File: test/e2e/httpproxy/007_client_cert_auth_test.go (hunk 1 above)
Comment (sunjayBhatia) [reply test/e2e/httpproxy/007_client_cert_auth_test.go:385]: 👍🏽 will update the other PR to cover this case

File: test/e2e/fixtures.go [hunk 2]
```diff
@@ -150,6 +154,151 @@ func (e *Echo) Deploy(ns, name string) func() {
...
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app.kubernetes.io/name": name},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "conformance-echo",
+							Image: "tsaarni/echoserver:latest",
+							Env: []corev1.EnvVar{
```
Comment (sunjayBhatia) [inline test/e2e/fixtures.go:197]: nit: missing from the original one but might be good to add these as well like the basic echo server https://github.com/projectcontour/contour/blob/eb8ff868c365af1f1ef83a31fcfa488f894253b6/test/e2e/fixtures.go#L86-L93

**Review by sunjayBhatia — APPROVED** [review]

File: test/e2e/fixtures.go (hunk 2 above)
Comment (skriss) [reply test/e2e/fixtures.go:197]: Oh yeah, good call, I did that somewhere, let me find it..

**Pull request merged**